go/libkb/api_test.go

```diff
@@ -24,3 +26,143 @@ func TestIsReddit(t *testing.T) {
 		t.Fatal("should NOT be a reddit URL")
 	}
 }
+
+const (
+	uriExpected  = "https://api.keybase.io"
+	pingExpected = "https://api.keybase.io/_/api/1.0/ping.json"
+)
+
+func TestProductionCA(t *testing.T) {
+	tc := SetupTest(t, "prod_ca")
+	defer tc.Cleanup()
+
+	t.Log("WARNING: setting run mode to production, be careful:")
+	tc.G.Env.Test.UseProductionRunMode = true
+
+	if tc.G.Env.GetServerURI() != uriExpected {
+		t.Fatalf("production server uri: %s, expected %s", tc.G.Env.GetServerURI(), uriExpected)
+	}
+
+	tc.G.ConfigureAPI()
+
+	// make sure endpoint is correct:
+	arg := APIArg{Endpoint: "ping"}
+	internal, ok := tc.G.API.(*InternalAPIEngine)
+	if !ok {
+		t.Fatal("failed to cast API to internal api engine")
+	}
+	url := internal.getURL(arg)
+	if url.String() != pingExpected {
+		t.Fatalf("api url: %s, expected %s", url, pingExpected)
+	}
+
+	_, err := tc.G.API.Post(arg)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = tc.G.API.Get(arg)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestProductionBadCA(t *testing.T) {
+	tc := SetupTest(t, "prod_ca")
+	defer tc.Cleanup()
+
+	t.Log("WARNING: setting run mode to production, be careful:")
+	tc.G.Env.Test.UseProductionRunMode = true
+
+	if tc.G.Env.GetServerURI() != uriExpected {
+		t.Fatalf("production server uri: %s, expected %s", tc.G.Env.GetServerURI(), uriExpected)
+	}
+
+	// change the api CA to one that api.keybase.io doesn't know:
+	BundledCAs["api.keybase.io"] = unknownCA
```

Comment:
probably the cleanest way to do this is to make `Env` fetch this as a test parameter, no? As we do some of the other parameters. That way you don't have to worry about crashing before you set it back.  Alternatively, you can just set it back with `defer`